enterprise/causal-clustering/src/test/java/org/neo4j/causalclustering/scenarios/MultiClusterRoutingIT.java

```diff
@@ -0,0 +1,241 @@
+    }
+
+    @Test
+    public void procedureCallsShouldReflectMembershipChanges() throws Exception
+    {
+        String dbName = getFirstDbName( dbNames );
+        CoreClusterMember follower = cluster.getDbWithAnyRole( dbName, Role.FOLLOWER );
+        int followerId = follower.serverId();
+
+        cluster.removeCoreMemberWithServerId( followerId );
+
+        CoreGraphDatabase db = cluster.getDbWithAnyRole( dbName, Role.FOLLOWER, Role.LEADER ).database();
+
+        Function<CoreGraphDatabase, Set<Endpoint>> getResult = database ->
+        {
+            Optional<MultiClusterRoutingResult> optResult = callProcedure( database, GET_SUPER_CLUSTER_ROUTERS, Collections.emptyMap() );
+
+            return optResult.map( r ->
+                    r.routers().values().stream()
+                            .flatMap( List::stream )
+                            .collect( Collectors.toSet() )
+            ).orElse( Collections.emptySet() );
+        };
+
+        assertEventually( "The procedure should return one fewer routers when a core member has been removed.",
+                () -> getResult.apply( db ).size(), is(numCores - 1 ), 15, TimeUnit.SECONDS );
+
+        BiPredicate<Set<Endpoint>, CoreClusterMember> containsFollower = ( rs, f ) ->
+                rs.stream().anyMatch( r -> r.address().toString().equals( f.boltAdvertisedAddress() ) );
+
+        assertEventually( "The procedure should not return a host as a router after it has been removed from the cluster",
+                () -> containsFollower.test( getResult.apply( db ), follower ), is( false ), 15, TimeUnit.SECONDS );
+
+        CoreClusterMember newFollower = cluster.addCoreMemberWithId( followerId );
+        newFollower.start();
+
+        assertEventually( "The procedure should return one more router when a core member has been added.",
+                () -> getResult.apply( db ).size(), is( numCores ), 15, TimeUnit.SECONDS );
+        assertEventually( "The procedure should return a core member as a router after it has been added to the cluster",
+                () -> containsFollower.test( getResult.apply( db ), newFollower ), is( true ), 15, TimeUnit.SECONDS );
+
+    }
+
+    private static String getFirstDbName( Set<String> dbNames )
+    {
+        return dbNames.stream()
+                .findFirst()
+                .orElseThrow( () -> new IllegalArgumentException( "The dbNames parameter must not be empty." ) );
+    }
+
+    private static Optional<MultiClusterRoutingResult> callProcedure( CoreGraphDatabase db, ProcedureNames procedure, Map<String,Object> params )
+    {
+
+        Optional<MultiClusterRoutingResult> routingResult = Optional.empty();
+        try (
+                InternalTransaction tx = db.beginTransaction( KernelTransaction.Type.explicit, EnterpriseLoginContext.AUTH_DISABLED );
+                Result result = db.execute( tx, "CALL " + procedure.callName(), ValueUtils.asMapValue( params )) )
```

Comment:
Do you really want the db.execute in the twr? Seems like it might be better placed in the block.

Reply:
`Result` is `AutoCloseable`, and the twr will call `.close()` in reverse order (i.e. will close the `Result` first) as expected, so I thought it was better to let it handle everything. No strong opinion though - so can change for clarity if desired?

Reply:
My mistake. It's good as is.